### DISCUSSION TITLE
Creating a DTO incorrectly causes process to die; no errors anywhere to suggest why.

### DISCUSSION THREAD

**Discussion by maxanstey-acquire** [spatie/laravel-data#874] [Q&A]
Hi there,

I was running a test and got an error like this:

```
root@branch-affect-dev:/var/www/html# php artisan test --filter=testSchoolIsCreated
   Symfony\Component\Process\Exception\ProcessSignaledException 

  The process has been signaled with signal "11".

  at vendor/symfony/process/Process.php:461
    457▕             usleep(1000);
    458▕         }
```

I traced it my DTO below:

```
use Spatie\LaravelData\Dto;

class CreateSchoolData extends Dto
{
    public function __construct(
        public readonly string $schoolCode,
        public readonly string|null $name,
    ) {
    }
```

I reproduced the creation via tinker and the process just dies without logging anywhere:

```
root@branch-affect-dev:/var/www/html# php -d display_errors=1 -d error_reporting=E_ALL artisan tinker
Psy Shell v0.12.4 (PHP 8.3.11 — cli) by Justin Hileman
> \App\Data\CreateSchoolData::from(['school_code' => 'A123', 'name' => 'Some School'])
root@branch-affect-dev:/var/www/html# cat storage/logs/laravel.log 
root@branch-affect-dev:/var/www/html# 
```

Now the problem of course is that I needed to add a SnakeCaseMapper to the DTO but it wasn't ideal that nothing anywhere told me the creation of the DTO was the issue. I'd have expected an exception at least. Am I doing something wrong? Is there a way to get an exception to be thrown rather than it behaving as it does currently?

Thanks

**Comment by maxanstey-acquire** [comment]
I still think this behaviour is undesirable, but if I set `'validation_strategy' => \Spatie\LaravelData\Support\Creation\ValidationStrategy::Always->value` then I do get an exception, so I'd say my requirement is met at least. Thanks

**Comment by gigabites19** [comment]
I'm facing the same issue. It does not die within tinker session and throws `Illuminate\Validation\ValidationException`, but it does when I try to create invalid data object in a controller method and my browser shows the following: `The page isn’t redirecting properly`.

My config is default except for: `'validation_strategy' => \Spatie\LaravelData\Support\Creation\ValidationStrategy::Always->value`
